### DISCUSSION TITLE
Is widevine support planned in the future for kernel 6.1?

### DISCUSSION THREAD

**Discussion by fenree** [Joshua-Riek/ubuntu-rockchip#788] [Q&A]
What the title says pretty much.

**Comment by develperbayman** [comment]
let me field this .....probably its only a matter of time because that work is/has been done for the previous kernel i do beileve

**Comment by nrpetonr** [comment]
plus one here. how may we support?

**Comment by nrpetonr** [comment]
use this [script ](https://github.com/AsahiLinux/widevine-installer)followed by [browser spoofing,](https://chromewebstore.google.com/detail/user-agent-switcher-and-m/bhchdcejhohfmigjafbampogmaanbfkg?hl=en) to fix the problem with drm sources. enjoy...

**Reply by wmport** [reply]
I wonder if it will work under android 12, there kernel 6.1 ?

**Reply by nrpetonr** [reply]
i’d expect it is not even needed on android if using apps from official play store.
my experience running android under redroid container on ubuntu was that widevine was running fine.

**Comment by wmport** [comment]
In the current state, 24.04, I wouldn't say that widevine works directly, the contet processing is handled by rkmpp... it's hard to call this processing good. 

mpv https://edge1..../dash-live2/streams/ .... dash.mpd
 (+) Video --vid=1 (*) (h264 1280x720 25.000fps) (3800 kbps)
     Video --vid=2 (*) (h264 1280x720 25.000fps) (2400 kbps)
     Video --vid=3 (*) (h264 1280x720) (1250 kbps)
     Video --vid=4 (*) (h264 640x360) (950 kbps)
     Video --vid=5 (*) (h264 640x360) (350 kbps)
     Audio --aid=1 --alang=rus (*) (aac 2ch 48000Hz) (64 kbps)
 (+) Audio --aid=2 --alang=rus (*) (aac 2ch 48000Hz) (128 kbps)
Using hardware decoding (rkmpp).
rga_api version 1.10.0_[8]
AO: [pipewire] 48000Hz stereo 2ch floatp
VO: [gpu] 1280x720 drm_prime[nv12]
AV: 00:00:10 / 00:00:53 (19%) A-V:  0.000 Cache: 33s/9MB

Audio/Video desynchronisation detected! Possible reasons include too slow
hardware, temporary CPU spikes, broken drivers, and broken files. Audio
position will not match to the video (see A-V status field).

AV: 00:00:38 / 00:01:34 (40%) A-V:  0.000 ct:  9.997 Dropped: 596 Cache: 56s/14M
[ffmpeg] https: HTTP error 404 Not Found
AV: 00:00:38 / 00:01:34 (41%) A-V:  0.000 ct:  9.997 Dropped: 596 Cache: 56s/14M
[ffmpeg/demuxer] dash: Failed to open fragment of playlist
AV: 00:00:38 / 00:01:34 (41%) A-V:  0.000 ct:  9.997 Dropped: 596 Cache: 55s/14M
[ffmpeg] https: HTTP error 404 Not Found
AV: 00:00:39 / 00:01:34 (41%) A-V:  0.000 ct:  9.997 Dropped: 596 Cache: 55s/14M

2 or 3 minutes of playback and it's a complete failure......
.
In android using nanopc-t6 and https://play.google.com/store/apps/details?id=com.bitmovin.demo.player
and custom widevine there is not a single error in processing dash widevine stream content

**Reply by nrpetonr** [reply]
no problem for me on 24.04 with mentioned script.
i've been watching netflix show as typing this more than 15 minutes w/o any  hickups. same experience with disney+.
actually i can watch both of them in same time w/o any problem. each core utilized  between  10% to 30%.
